BotDatabase.py

```diff
@@ -80,30 +90,48 @@ def CleanupBackups(self):
                 
     ### Adding/Updating/Removing Server Entries ###
     def AddBotGuilds(self, ListOwnerAndServerTuples, BotID:int):
-        BotAdditionUpdates = []
+        BotAdditionUpdates:list[Server] = []
         for Entry in ListOwnerAndServerTuples:
-            BotAdditionUpdates.append(Entry + (-1, 0, BotID))
+            server = Server(
+                bot_instance_id = BotID,
+                discord_server_id = Entry.id,
+                owner_discord_user_id = Entry.owner_id,
+                activator_discord_user_id = Entry.owner_id
+            )
+            BotAdditionUpdates.append(server)
         
-        self.Database.executemany("INSERT INTO servers VALUES(?, ?, ?, ?, ?)", BotAdditionUpdates)
+        self.Database.bulk_save_objects(BotAdditionUpdates)
         self.Database.commit()
+
         Logger.Log(LogLevel.Notice, f"Bot #{BotID} had {len(BotAdditionUpdates)} new server updates")
         
     def SetNewServerOwner(self, ServerId:int, NewOwnerId:int, BotId:int):
-        ActivationChanges = []
-        ActivationChanges.append({"Id": ServerId, "OwnerId": NewOwnerId, "BotId": BotId})
-        self.Database.executemany("UPDATE servers SET OwnerId=:OwnerId WHERE Id=:Id AND Instance=:BotId", ActivationChanges)
-        self.Database.commit()
+        stmt = select(Server).where((Server.discord_server_id==ServerId) & (Server.bot_instance_id==BotId))
+        server = self.Database.scalars(stmt).first()
+
+        if (server is None):
+            Logger.Log(LogLevel.Warn, f"Bot had attempted to set new owner on non-existant server ")
+            return
         
+        server.owner_discord_user_id = NewOwnerId
+
+        self.Database.add(server)
+        self.Database.commit()
+           
     def RemoveServerEntry(self, ServerId:int, BotId:int):
-        if (self.IsInServer(ServerId)):  
-            self.Database.execute(f"DELETE FROM servers WHERE Id={ServerId} AND Instance={BotId}")
-            self.Database.commit()
-        else:
-            Logger.Log(LogLevel.Warn, f"Attempted to remove server {ServerId} but we are not in that list!")
-    
-    def SetBotActivationForOwner(self, Servers, IsActive:bool, BotId:int, OwnerId:int=-1, ActivatorId:int=-1):
-        ActivationChanges = []
-        ActivationAdditions = []
+        stmt = select(Server).where((Server.discord_server_id==ServerId) & (Server.bot_instance_id==BotId))
+        server = self.Database.scalars(stmt).first()
+
+        if (server is None):
+            Logger.Log(LogLevel.Warn, f"Attempted to remove an invalid server id: {ServerId}")
+            return
+        
+        self.Database.delete(server)
+        self.Database.commit()
+
+    def SetBotActivationForOwner(self, Servers:list[int], IsActive:bool, BotId:int, OwnerId:int=-1, ActivatorId:int=-1):
+        ActivationChanges:list[int] = []
```

Comment:
Can we make instead a count of the changes rather than pushing lists of server ids?

Reply:
It was originally a one to one change to your original code, merely adding the type cast hint; but it has been changed and code flow was shortened